test/message-list.test.js

```diff
@@ -136,5 +153,141 @@ describe('message-list', () => {
+      const thirdMessage = messageList.shadowRoot.querySelectorAll('vaadin-message')[2];
+
+      // click on third item to give it tabindex=0
+      thirdMessage.dispatchEvent(new CustomEvent('mousedown', { composed: true, bubbles: true }));
+      thirdMessage.dispatchEvent(new CustomEvent('focus', { composed: true, bubbles: true }));
+      thirdMessage.dispatchEvent(new CustomEvent('mouseup', { composed: true, bubbles: true }));
+
+      // set message list to shorter than three items, so that tabIndex=0 can't be maintained on third item
+      messageList.items = [
+        {
+          text: 'This is a new list',
+          time: '2:35 PM',
+          user: {
+            name: 'Steve Mops',
+            abbr: 'SM',
+            colorIndex: 2
+          }
+        },
+        {
+          text: 'With two items',
+          time: '2:35 PM',
+          user: {
+            name: 'Steve Mops',
+            abbr: 'SM',
+            colorIndex: 2
+          }
+        }
+      ];
+      await nextRender(messageList);
+      const firstMessage = messageList.shadowRoot.querySelectorAll('vaadin-message')[0];
+      // Verify that the first item got the new tabIndex=0.
+      expect(firstMessage.tabIndex).to.be.equal(0);
+    });
+  });
+
+  describe('mouse navigation', () => {
+    let messageElements;
+    beforeEach(async () => {
+      messageList.items = messages;
+      await nextRender(messageList);
+      messageElements = messageList.shadowRoot.querySelectorAll('vaadin-message');
+    });
+
+    it('click message to give it focus', () => {
+      messageElements[1].dispatchEvent(new CustomEvent('mousedown', { composed: true, bubbles: true }));
+      messageElements[1].dispatchEvent(new CustomEvent('focus', { composed: true, bubbles: true }));
+      messageElements[1].dispatchEvent(new CustomEvent('mouseup', { composed: true, bubbles: true }));
+
+      verifyHasFocus(messageElements, messageElements[1]);
+      expect(messageElements[1].hasAttribute('focus-ring')).to.be.false;
+      expect(messageElements[1].tabIndex).to.be.equal(0);
+    });
+  });
+
+  describe('keyboard navigation', () => {
+    let messageElements;
+    beforeEach(async () => {
+      messageList.items = messages;
+      await nextRender(messageList);
+      messageElements = messageList.shadowRoot.querySelectorAll('vaadin-message');
+    });
+
+    function arrowDown(element) {
+      keyDownOn(element, 40, [], 'ArrowDown');
+      keyUpOn(element, 40, [], 'ArrowDown');
+    }
+
+    function arrowUp(element) {
+      keyDownOn(element, 38, [], 'ArrowUp');
+      keyUpOn(element, 38, [], 'ArrowUp');
+    }
+
+    function home(element) {
+      keyDownOn(element, 36, [], 'Home');
+      keyUpOn(element, 36, [], 'Home');
+    }
+
+    function end(element) {
+      keyDownOn(element, 35, [], 'End');
+      keyUpOn(element, 35, [], 'End');
+    }
+
+    it('no focus before interaction', () => {
+      verifyNoFocus(messageElements);
```

Comment:
Please use explicit assertion instead of a helper:

```js
messageElements.forEach((message) => {
  expect(message.hasAttribute('focused')).to.be.false;
});
```

Same applies to other tests below.

Reply:
Done.